workflows/mag.nf

```diff
@@ -223,13 +225,45 @@ workflow MAG {
     )
     ch_versions = ch_versions.mix(FASTQC_RAW.out.versions.first())
 
-    FASTP (
-        ch_raw_short_reads,
-        [],
-        []
-    )
-    ch_short_reads = FASTP.out.reads
-    ch_versions = ch_versions.mix(FASTP.out.versions.first())
+    if ( params.clip_tool == 'fastp' ) {
+        ch_clipmerge_out = FASTP (
+            ch_raw_short_reads,
+            params.fastp_save_trimmed_fail,
+            []
+        )
+        ch_short_reads = FASTP.out.reads
+        ch_versions = ch_versions.mix(FASTP.out.versions.first())
+
+    } else if ( params.clip_tool == 'adapterremoval' ) {
+
+        // due to strange output file scheme in AR2, have to manually separate
+        // SE/PE to allow correct pulling of reads after.
+        ch_adapterremoval_in = ch_raw_short_reads
+            .branch {
+                    single: it[0]['single_end']
+                    paired: !it[0]['single_end']
+                }
+
+        ADAPTERREMOVAL_PE ( ch_adapterremoval_in.paired, [] )
+        ADAPTERREMOVAL_SE ( ch_adapterremoval_in.single, [] )
+
+        // pair1 and 2 come for PE data from separate output channels, so bring
+        // this back together again here
+        ch_adapterremoval_pe_out = Channel.empty()
+        ch_adapterremoval_pe_out = ADAPTERREMOVAL_PE.out.pair1_truncated
+            .join(ADAPTERREMOVAL_PE.out.pair2_truncated)
+            .dump(tag: "ar_pe_mix_out")
```

Comment:
If I didnt miss anything, you typically remove the `.dump()` usually?
```suggestion
```

Reply:
Yes correct, normally remove as it seems you guys don't leave it in.